Prevent execution environment from being assigned to a new organization

fosterseth opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
related https://github.com/ansible/awx/issues/9769
- ee organization can be changed to null (less restrictive)
- if organization is null, cannot be assigned to org (more restrictive)
- if org is assigned, it cannot be set to a different org

<!---
If you are fixing an existing issue, please include "related #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bugfix Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
 - API

##### AWX VERSION
<!--- Paste verbatim output from `make VERSION` between quotes below -->
```
awx: 19.0.0
```

nixocio commented:
I tested the following scenarios  - logged in as system admin.

1. Being able to assign an Org to a Globally Available EE. 
2. Not being able to remove an Org from an EE - to make the EE globally available.
3. Not being able to modify the Org already assigned to an EE.

I will work on the UI side to mimic those changes.

chrismeyersfsu commented on awx/api/serializers.py:
```diff
@@ -1412,6 +1412,18 @@ def get_related(self, obj):
+            if 'organization' in attrs:
+                org = attrs.get('organization', None)
+                if not org:
+                    raise serializers.ValidationError({"organization": _("Cannot remove the execution environment from the organization")})
```
Setting it to NULL should be OK. In this case you are moving from more restrictive to less restrictive.

Anyone can use EE w/ org Null.

jbradberry replied:
I concur.

fosterseth replied:
we currently allow going from null to non-null. So in essence a user could change from org-3 to org-2 by first setting to null, then setting to org-2.

chrismeyersfsu replied:
We shouldn't allow going from NULL to ORG1. Resources that point to the EE when it was NULL that are outside of ORG1 are now using the EE, outside of the org they are allowed to.

fosterseth replied:
thanks, changed it to prevent going from null to an org. But can go from org to null

fosterseth committed:
Prevent ee from being assigned to a new organization

- ee organization can be changed to null (less restrictive)
- if organization is null, cannot be assigned to org (more restrictive)
- if org is assigned, it cannot be set to a different org
